### PR TITLE
qt: fix minor issues in 6.x

### PR DESCRIPTION
**Pull request by fnadeau**
Specify library name and version:  **qt/6.5.3**

Fix issue #20924.

Note to reviewers:
- See #20817 for Svg
- Technically, some version of msvc 191 support c++17. Is there a better way to detect cpp standard?
- Qt6 support DirectX 9,11 and 12. Shoud we add d3d12?

Tested with
```
#include <QApplication>
#include <QPushButton>
#include <QtPlugin>
Q_IMPORT_PLUGIN (QWindowsIntegrationPlugin);

//#pragma comment(lib, "d3d9")
//#pragma comment(lib, "setupapi")
//#pragma comment(lib, "SHCore")

int main(int argc, char *argv[])
{
    QApplication a(argc, argv);

    QPushButton button ("Hello world!");
    button.show();

    return a.exec(); 
}
```
Pragmas are needed without patch
```
set(CMAKE_PROJECT_TOP_LEVEL_INCLUDES cmake/conan_provider.cmake)
cmake_minimum_required(VERSION 3.9)
project(FormatOutput CXX)

list(APPEND CMAKE_MODULE_PATH ${CMAKE_BINARY_DIR})
list(APPEND CMAKE_PREFIX_PATH ${CMAKE_BINARY_DIR})

set(CMAKE_BUILD_TYPE Release)

if(NOT EXISTS "${CMAKE_BINARY_DIR}/conan.cmake")
  message(STATUS "Downloading conan.cmake from https://github.com/conan-io/cmake-conan")
  file(DOWNLOAD "https://raw.githubusercontent.com/conan-io/cmake-conan/0.18.1/conan.cmake"
                "${CMAKE_BINARY_DIR}/conan.cmake"
                TLS_VERIFY ON)
endif()

find_package(Qt6 REQUIRED COMPONENTS Core Gui Widgets QWindowsIntegrationPlugin)
set(CMAKE_AUTOMOC ON)
set(CMAKE_AUTORCC ON)
set(CMAKE_AUTOUIC ON)

add_executable(main main.cpp)
target_link_libraries(main Qt6::Core Qt6::Gui Qt6::Widgets Qt6::QWindowsIntegrationPlugin)

target_compile_features(main PRIVATE cxx_std_17)
```

---

- [x] I've read the [contributing guidelines](https://github.com/conan-io/conan-center-index/blob/master/CONTRIBUTING.md).
- [x] I've used a [recent](https://github.com/conan-io/conan/releases/latest) Conan client version close to the [currently deployed](https://github.com/conan-io/conan-center-index/blob/master/.c3i/config_v1.yml#L6).
- [ ] I've tried at least one configuration locally with the [conan-center hook](https://github.com/conan-io/hooks.git) activated.
I'm using Conan 2.0, hooks don't work as per documentation.

### CODE REVIEW

**Comment by fnadeau** [comment]
> * [qt: add QT_NO_DEBUG define in release build #19631](https://github.com/conan-io/conan-center-index/pull/19631)
> * [qt/6.5.x: Only output a warning when using GCC 9 #19339](https://github.com/conan-io/conan-center-index/pull/19339)
> * [Add QuickPrivate module #19082](https://github.com/conan-io/conan-center-index/pull/19082)
> * [qt6: fixes for cross compiling #15678](https://github.com/conan-io/conan-center-index/pull/15678)

To best of my knowledge, this PR is not in conflict with the one above.

**Review by ericLemanissier — APPROVED** [review]

**Review by jwillikers — APPROVED** [review]

**Comment by fnadeau** [comment]
There are two reviewers, what else is missing?

**Review by MartinDelille — APPROVED** [review]

**Review by ericLemanissier — APPROVED** [review]
cf https://github.com/conan-io/conan-center-index/blob/master/docs/review_process.md#rule-of-2-reviews
You just have to wait for a review from a team member: `team` type in https://github.com/conan-io/conan-center-index/blob/master/.c3i/reviewers.yml

**Comment by fnadeau** [comment]
Thanks for clarifying the reviewers process. I was misled by these two sentences: "At least 2 approving reviews are required by reviewers with write access" and "Merging can be done automatically with 2 approving reviews." I assume these are generic GitHub messages.

**Review by uilianries — APPROVED** [review]

**Pull request merged**
